12.LOCO_BULK_MLP/0.data_splits/scripts/1.data_splits_model_multiclass.py

```diff
@@ -0,0 +1,335 @@
+    strict=True
+)
+treatment_splits_file_path = pathlib.Path(MLP_path / "splits.toml").resolve(strict=True)
+# read toml files
+ground_truth = toml.load(ground_truth_file_path)
+treatment_splits = toml.load(treatment_splits_file_path)
+
+
+# In[6]:
+
+
+# get information from toml files
+apoptosis_groups_list = ground_truth["Apoptosis"]["apoptosis_groups_list"]
+pyroptosis_groups_list = ground_truth["Pyroptosis"]["pyroptosis_groups_list"]
+healthy_groups_list = ground_truth["Healthy"]["healthy_groups_list"]
+test_split_100 = treatment_splits["splits"]["data_splits_100"]
+test_split_75 = treatment_splits["splits"]["data_splits_75"]
+
+
+# In[7]:
+
+
+np.random.seed(0)
+if mlp_params.DATA_SUBSET_OPTION == "True":
+    df1 = df1.groupby("oneb_Metadata_Treatment_Dose_Inhibitor_Dose").apply(
+        lambda x: x.sample(n=mlp_params.DATA_SUBSET_NUMBER, random_state=0)
+    )
+    print("Data Subset Is On")
+    print(f"Data is subset to {mlp_params.DATA_SUBSET_NUMBER} per treatment group")
+    print(df1.shape)
+    df1.reset_index(drop=True, inplace=True)
+else:
+    print("Data Subset Is Off")
+
+
+# In[8]:
+
+
+# add apoptosis, pyroptosis and healthy columns to dataframe
+df1["apoptosis"] = df1["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].isin(
+    apoptosis_groups_list
+)
+df1["pyroptosis"] = df1["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].isin(
+    pyroptosis_groups_list
+)
+df1["healthy"] = df1["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].isin(
+    healthy_groups_list
+)
+
+# merge apoptosis, pyroptosis, and healthy columns into one column
+conditions = [
+    (df1["apoptosis"] == True),
+    (df1["pyroptosis"] == True),
+    (df1["healthy"] == True),
+]
+choices = ["apoptosis", "pyroptosis", "healthy"]
+df1["labels"] = np.select(conditions, choices, default="healthy")
+
+# drop apoptosis, pyroptosis, and healthy columns
+df1.drop(columns=["apoptosis", "pyroptosis", "healthy"], inplace=True)
+
+
+# ### Split said data
+
+# In[9]:
+
+
+# randomly select wells to hold out for testing one per treatment group
+# stratified by treatment group
+np.random.seed(seed=0)
+wells_to_hold = (
+    df1.groupby("oneb_Metadata_Treatment_Dose_Inhibitor_Dose")
+    .agg(np.random.choice)["Metadata_Well"]
+    .to_list()
+)
+df_holdout = df1[df1["Metadata_Well"].isin(wells_to_hold)]
+df = df1[~df1["Metadata_Well"].isin(wells_to_hold)]
+
+
+print("Wells held out for testing:", df_holdout["Metadata_Well"].unique())
+print(
+    "Wells to use for training, validation, and testing", df1["Metadata_Well"].unique()
+)
+print(df_holdout.shape, df.shape)
+
+
+# In[10]:
+
+
+# variable test and train set splits
+# 100% test set
+# subset the following treatments for test set
+treatment_holdout = df[
+    df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].isin(test_split_100)
+]
+df = df[~df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].isin(test_split_100)]
+print(treatment_holdout.shape, df.shape)
```

Comment:
I am a bit confused here, can you provide me a bit more context behind what is being subset here?

Reply:
the treatments applied to the wells! :D